### PR TITLE
Allow non-critical Spotify API requests to fail without breaking the pipeline

### LINKED ISSUE #28: Import stops when 404
Importer stops when retrieving a 404 http status (e.g. spotify returns 404 if their data processor is delayed and there is no data yet)

```
 Starting Spotify connector
Launching connector.
Initializing environment
Done initializing environment
Traceback (most recent call last):
  File "/home/runner/work/pipelines/pipelines/spotify/main.py", line 305, in <module>
    main()
  File "/home/runner/work/pipelines/pipelines/spotify/main.py", line 190, in main
    fetch_and_capture(
  File "/home/runner/work/pipelines/pipelines/spotify/main.py", line 103, in fetch_and_capture
    raise e
  File "/home/runner/work/pipelines/pipelines/spotify/main.py", line 93, in fetch_and_capture
    data = connector_call()
  File "/home/runner/work/pipelines/pipelines/spotify/main.py", line 193, in <lambda>
    lambda: spotify_connector.listeners(start, end),
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/spotifyconnector/connector.py", line 322, in listeners
    return self._request(url, params=self._date_params(start, end))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/spotifyconnector/connector.py", line 221, in _request
    response.raise_for_status()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://generic.wg.spotify.com/podcasters/v0/shows/***/listeners?start=2023-01-13&end=2023-01-16
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by mre**
Fixes #28

### CODE REVIEW

**Comment by mre** [comment]
Test run: https://github.com/openpodcast/pipelines/actions/runs/3929831807/jobs/6719107791

**Comment by mre** [comment]
The test run succeeded. There are still the expected 404 errors, but the pipeline continues to run and imports the remaining data. I think we can merge this.

File: spotify/main.py [hunk 1]
```diff
@@ -183,6 +183,7 @@ def main():
         open_podcast_client,
         start,
         end,
+        fallible=True,
```
Comment (woolfg) [inline spotify/main.py:186]: I would suggest something more informative such as `continueOnError` or `doNotRaiseErrors`. Even though I would aso suggest to log the error and not just silently return. Can be a just q quick logline like "404 on URL..."

**Review by woolfg — APPROVED** [review]
Looks good, I would just add the log line and change the var name

File: spotify/main.py (hunk 1 above)
Comment (mre) [reply spotify/main.py]: Good idea. Renamed to `continueOnError`.

Comment (mre) [reply spotify/main.py]: Good idea. Renamed to `continueOnError`.

**Comment by mre** [comment]
Added a log line. 👍 
Will do another test run and merge it if it passes.

**Comment by mre** [comment]
https://github.com/openpodcast/pipelines/actions/runs/3930488982

**Comment by mre** [comment]
Added log message:
```
2023-01-16 13:08:34.131 | ERROR    | __main__:fetch_and_capture:107 - Encountered a non-critical error while fetching performance: 404 Client Error: Not Found for url: https://generic.wg.spotify.com/podcasters/v0/episodes/4XDF4l1KLQVmXAKMM3JgnV/performance.Maybe the data is not available yet.Continuing because continueOnError is set to True
```

CI/CD succeeded; merging.

**Pull request merged**